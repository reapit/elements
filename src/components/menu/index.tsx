export * from './menu'
export * from './menu.atoms'
export * from './styles'
