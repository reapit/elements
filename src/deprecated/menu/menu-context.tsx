import { createContext, ReactNode, useContext, type FC } from 'react'
import { useDeprecatedMenu } from './use-menu'

/** @deprecated */
interface DeprecatedMenuProviderProps {
  children: ReactNode
}

/** @deprecated */
export const DeprecatedMenuContext = createContext<useDeprecatedMenu | null>(null)

/** @deprecated */
export const DeprecatedMenuProvider: FC<DeprecatedMenuProviderProps> = ({ children }) => {
  const menu = useDeprecatedMenu()

  return <DeprecatedMenuContext.Provider value={menu}>{children}</DeprecatedMenuContext.Provider>
}

/** @deprecated */
export const useDeprecatedMenuContext = () => {
  const context = useContext(DeprecatedMenuContext)
  if (!context) {
    throw new Error('useMenuContext must be used within a MenuProvider')
  }
  return context
}
