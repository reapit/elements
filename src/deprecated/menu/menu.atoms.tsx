import type {
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  FC,
  HTMLAttributes,
  MouseEventHandler,
  ReactNode,
} from 'react'
import {
  ElDeprecatedMenuItemAnchor,
  ElDeprecatedMenuItemButton,
  ElDeprecatedMenuItemGroup,
  ElDeprecatedMenuItemGroupList,
  ElDeprecatedMenuItemGroupTitle,
  ElDeprecatedMenuList,
} from './styles'
import type { sizeType } from '../../types/core'

/** @deprecated */
interface CommonDeprecatedMenuItemProps {
  children?: ReactNode
  /**
   * Whether the menu is closed when clicking this item
   *
   * @example
   * <MenuItem closeMenu={false}>This won't close the menu</MenuItem>
   *
   * @default true
   */
  closeMenu?: boolean
  /**
   * Whether the Menu item is active
   *
   * @default false
   **/
  isActive?: boolean
}

/** @deprecated */
interface DeprecatedMenuItemAsButtonElementProps
  extends CommonDeprecatedMenuItemProps,
    ButtonHTMLAttributes<HTMLButtonElement> {
  href?: never
  disabled?: boolean
  onClick?: MouseEventHandler<HTMLButtonElement>
}

/** @deprecated */
interface DeprecatedMenuItemAsAnchorElementProps
  extends CommonDeprecatedMenuItemProps,
    AnchorHTMLAttributes<HTMLAnchorElement> {
  /** MenuItemAsAnchor currently doesn't support disabled state, use MenuItemButton instead */
  disabled?: never
}

/** @deprecated */
export type DeprecatedMenuItemContainerProps =
  | DeprecatedMenuItemAsButtonElementProps
  | DeprecatedMenuItemAsAnchorElementProps

/** @deprecated */
export interface DeprecatedMenuListProps extends HTMLAttributes<HTMLDivElement> {
  maxWidth?: sizeType
  maxHeight?: sizeType
}

/**
 * The `MenuItemGroup` component is a wrapper for `MenuItem` which has optional label
 * can be used for grouping multiple type of list item e.g RadioItem (in future).
 *
 * @deprecated
 */
export const DeprecatedMenuItemGroup: FC<
  HTMLAttributes<HTMLDivElement> & {
    label?: string
    maxHeight?: sizeType
  }
> = ({ children, label, maxHeight, ...rest }) => {
  return (
    <ElDeprecatedMenuItemGroup {...rest} role="group">
      {!!label && <ElDeprecatedMenuItemGroupTitle>{label}</ElDeprecatedMenuItemGroupTitle>}
      <ElDeprecatedMenuItemGroupList style={{ ...rest?.style, maxHeight: `var(${maxHeight})` }}>
        {children}
      </ElDeprecatedMenuItemGroupList>
    </ElDeprecatedMenuItemGroup>
  )
}

/** @deprecated */
export const DeprecatedMenuItemContainer: FC<DeprecatedMenuItemContainerProps> = ({
  children,
  isActive,
  disabled,
  closeMenu = true,
  ...rest
}) => {
  if (!isItemAsButtonElement(rest)) {
    return (
      <ElDeprecatedMenuItemAnchor
        {...(rest as DeprecatedMenuItemAsAnchorElementProps)}
        role="menuitem"
        data-close-menu={closeMenu || !!disabled}
        aria-current={isActive ? 'page' : undefined}
      >
        {children}
      </ElDeprecatedMenuItemAnchor>
    )
  }

  return (
    <ElDeprecatedMenuItemButton
      {...rest}
      role="menuitem"
      data-close-menu={closeMenu}
      disabled={disabled}
      aria-disabled={disabled}
      aria-current={isActive ? 'true' : undefined}
      tabIndex={disabled ? -1 : 0}
    >
      {children}
    </ElDeprecatedMenuItemButton>
  )
}

/** @deprecated */
export const DeprecatedMenuList: FC<DeprecatedMenuListProps> = ({ children, maxWidth, maxHeight, ...rest }) => (
  <ElDeprecatedMenuList
    {...rest}
    style={{ ...rest?.style, maxWidth: `var(${maxWidth})`, maxHeight: `var(${maxHeight})` }}
    data-has-max-width={!!maxWidth}
    role="menu"
  >
    {children}
  </ElDeprecatedMenuList>
)

function isItemAsButtonElement(
  props: DeprecatedMenuItemContainerProps,
): props is DeprecatedMenuItemAsButtonElementProps {
  return !props.href
}
