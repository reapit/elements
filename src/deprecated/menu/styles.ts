import { css } from '@linaria/core'
import { styled } from '@linaria/react'

/** @deprecated */
export const ElDeprecatedMenuPopover = styled.div`
  position: absolute;
  z-index: 100;
`

/** @deprecated */
export const ElDeprecatedMenu = styled.div`
  position: relative;

  &[data-alignment='left'] {
    > ${ElDeprecatedMenuPopover} {
      left: 0;
    }
  }
  &[data-alignment='right'] {
    > ${ElDeprecatedMenuPopover} {
      right: 0;
    }
  }
`

/** @deprecated */
export const ElDeprecatedMenuItemSupplementaryInfo = styled.span`
  color: var(--comp-menu-colour-text-hover-secondary);
  font-family: var(--font-xs-regular-family);
  font-size: var(--font-xs-regular-size);
  font-weight: var(--font-xs-regular-weight);
  line-height: var(--font-xs-regular-line_height);
  letter-spacing: var(--font-xs-regular-letter_spacing);
`

/** @deprecated */
export const ElDeprecatedMenuItemLabelContainer = styled.span`
  text-align: left;
`

/** @deprecated */
export const ElDeprecatedMenuItemLabel = styled.span`
  font-family: var(--font-sm-regular-family);
  font-size: var(--font-sm-regular-size);
  font-weight: var(--font-sm-regular-weight);
  line-height: var(--font-sm-regular-line_height);
  letter-spacing: var(--font-sm-regular-letter_spacing);
  text-align: left;
  white-space: nowrap;
  margin-right: var(--spacing-2);
`

/** @deprecated */
export const ElDeprecatedMenuItemContent = styled.div`
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  gap: var(--spacing-1);
  flex: 1 0 0;
`

/** @deprecated */
export const ElDeprecatedMenuItemIcon = styled.div`
  &,
  svg {
    width: var(--icon_size-m);
    height: var(--icon_size-m);
    color: var(--comp-menu-colour-icon-default-right);
  }
`

/** @deprecated */
export const elDeprecatedMenuItemLeftIcon = css`
  &,
  svg {
    color: var(--comp-menu-colour-icon-default-left);
  }
`

const baseMenuItemStyles = `
  width: 100%;
  cursor: pointer;
  display: flex;
  align-items: flex-start;
  padding: var(--spacing-3);
  gap: var(--spacing-3);
  border-radius: var(--comp-menu-border-radius);
  background: var(--comp-menu-colour-fill-default);
  color: var(--comp-menu-colour-text-default-primary);
  text-decoration: none;

  &:hover {
    background: var(--comp-menu-colour-fill-hover);
  }

  &:focus-visible {
    outline: none;
  }

  &:focus {
    outline: var(--border-width-double) solid var(--colour-border-focus);

    /* prevent the outline to be hidden behind next item */
    position: relative;
    z-index: 1;
  }

  &[aria-current="true"], &[aria-current="page"] {
    background: var(--comp-menu-colour-fill-highlighted);
  }

  &[aria-disabled="true"] {
    cursor: not-allowed;
    background: var(--comp-menu-colour-fill-default);
    color: var(--comp-menu-colour-text-disabled);
  }
`

/** @deprecated */
export const ElDeprecatedMenuItemButton = styled.button`
  ${baseMenuItemStyles}
  &[aria-current="true"], &[aria-current="page"] {
    ${ElDeprecatedMenuItemLabel} {
      color: var(--comp-menu-colour-text-default-action);
    }

    .${elDeprecatedMenuItemLeftIcon} {
      &,
      svg {
        color: var(--comp-menu-colour-icon-default-action);
      }
    }
  }

  &[aria-disabled='true'] {
    ${ElDeprecatedMenuItemLabel} {
      color: var(--comp-menu-colour-text-disabled-primary);
    }
  }

  /* customize html button */
  background-color: transparent;
  border: none;
`

/** @deprecated */
export const ElDeprecatedMenuItemAnchor = styled.a`
  ${baseMenuItemStyles}
  &[aria-current="true"], &[aria-current="page"] {
    ${ElDeprecatedMenuItemLabel} {
      color: var(--comp-menu-colour-text-default-action) !important;
    }
  }
`

/** @deprecated */
export const ElDeprecatedMenuItemGroupTitle = styled.div`
  font-family: var(--font-family);
  font-size: var(--font-size-2xs);
  font-style: normal;
  font-weight: var(--font-weight-semibold);
  line-height: var(--line-height-2xs);
  letter-spacing: var(--letter-spacing-2xs);
  color: var(--text-placeholder);
  text-transform: uppercase;
  display: flex;
  height: 32px;
  padding: var(--spacing-none) var(--spacing-4);
  align-items: center;
  align-self: stretch;
`
/** @deprecated */
export const ElDeprecatedMenuItemGroup = styled.div`
  padding-inline: var(--spacing-2);
`

/** @deprecated */
export const ElDeprecatedMenuItemGroupList = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  overflow: auto;
`

/** @deprecated */
export const ElDeprecatedMenuList = styled.div`
  width: max-content;
  padding: var(--spacing-2) 0;
  border-radius: var(--corner-default);
  background-color: var(--fill-white);
  box-shadow: 0 4px 16px 0 #222b3329;
  overflow: auto;

  &[data-has-max-width='true'] {
    ${ElDeprecatedMenuItemLabel} {
      white-space: normal;
    }
  }

  ${ElDeprecatedMenuItemGroup}:not(:last-child) {
    border-bottom: 1px solid var(--comp-divider-colour-border-solid);
    padding-bottom: var(--spacing-2);
    margin-bottom: var(--spacing-2);
  }
`
