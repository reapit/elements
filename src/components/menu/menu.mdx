import { Canvas, Meta, Controls } from '@storybook/blocks'
import * as MenuStories from './menu.stories'
import { RenderHtmlMarkup } from '../../storybook/render-html-markup'

<Meta of={MenuStories} />

# Menu

The Menu component is a panel used to display a list of buttons and/or anchors which can be ordered in Menu Groups

Each Menu Group can have a title shown above the list

More features will be added to Menu in the future

<Controls />

## Default

By default, the `Menu` is left-aligned, but this can be adjusted using the `data-alignment`. Additionally (for React user), the vertical position of the Menu is automatically adjusted based on the available space below the trigger and `window` each time the menu is opened (can use `yOffset` to have more adjustment).

`closeMenu`={false} can be set to keep the menu open when it is clicked.

Note: `closeMenu` and `yOffset` are react only props.

<Canvas of={MenuStories.Default} />

<RenderHtmlMarkup of={MenuStories.Default} />

## With Alignment

Example for `Menu` with right Alignment.

<Canvas of={MenuStories.WithCustomAlignment} />

<RenderHtmlMarkup of={MenuStories.WithCustomAlignment} />
