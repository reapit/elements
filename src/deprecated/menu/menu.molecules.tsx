import type { FC, ReactNode } from 'react'
import { DeprecatedMenuItemContainer, type DeprecatedMenuItemContainerProps } from './menu.atoms'
import {
  ElDeprecatedMenuItemContent,
  ElDeprecatedMenuItemIcon,
  ElDeprecatedMenuItemLabel,
  ElDeprecatedMenuItemLabelContainer,
  elDeprecatedMenuItemLeftIcon,
  ElDeprecatedMenuItemSupplementaryInfo,
} from './styles'

/** @deprecated */
export interface DeprecatedMenuItemProps extends Omit<DeprecatedMenuItemContainerProps, 'children'> {
  label: string
  supplementaryInfo?: string
  badge?: ReactNode
  leftIcon?: ReactNode
  rightIcon?: ReactNode
}

/** @deprecated */
export const DeprecatedMenuItem: FC<DeprecatedMenuItemProps> = ({
  label,
  supplementaryInfo,
  badge,
  leftIcon,
  rightIcon,
  ...props
}) => {
  return (
    <DeprecatedMenuItemContainer {...(props as DeprecatedMenuItemContainerProps)}>
      {leftIcon && (
        <ElDeprecatedMenuItemIcon className={elDeprecatedMenuItemLeftIcon}>{leftIcon}</ElDeprecatedMenuItemIcon>
      )}
      <ElDeprecatedMenuItemContent>
        <ElDeprecatedMenuItemLabelContainer>
          <ElDeprecatedMenuItemLabel>{label}</ElDeprecatedMenuItemLabel>
          {badge}
        </ElDeprecatedMenuItemLabelContainer>
        {supplementaryInfo && (
          <ElDeprecatedMenuItemSupplementaryInfo>{supplementaryInfo}</ElDeprecatedMenuItemSupplementaryInfo>
        )}
      </ElDeprecatedMenuItemContent>
      {rightIcon && <ElDeprecatedMenuItemIcon>{rightIcon}</ElDeprecatedMenuItemIcon>}
    </DeprecatedMenuItemContainer>
  )
}
