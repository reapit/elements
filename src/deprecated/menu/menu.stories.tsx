import type { Meta, StoryObj } from '@storybook/react-vite'
import { useEffect, type ComponentProps } from 'react'
import { DeprecatedMenu, useDeprecatedMenuContext } from '.'
import { DeprecatedButton } from '../button'
import { DeprecatedIcon } from '../icon'
import { FlexContainer, MainContainer } from '../layout'
import { elHScreen } from '../../styles/deprecated-sizing'
import { Badge } from '../../core/badge'

const meta: Meta<typeof DeprecatedMenu> = {
  title: 'Deprecated/DeprecatedMenu',
}

export default meta
type Story = StoryObj<typeof DeprecatedMenu>

export const Default: StoryObj = {
  render: () => {
    return (
      <DeprecatedMenu>
        <DeprecatedMenu.Trigger>
          {({ getTriggerProps }) => (
            <DeprecatedButton {...getTriggerProps()} iconLeft={<DeprecatedIcon icon="more" fontSize="1rem" />} />
          )}
        </DeprecatedMenu.Trigger>
        <DeprecatedMenu.Popover>
          <DeprecatedMenu.List>
            <DeprecatedMenu.Group label="Group Title">
              <DeprecatedMenu.Item label="Menu Item" onClick={console.log} />
              <DeprecatedMenu.Item label="Menu Item as anchor" href="/#" />
              <DeprecatedMenu.Item label="Menu Item (keep open)" closeMenu={false} isActive />
              <DeprecatedMenu.Item label="Menu Item (disabled)" onClick={console.log} disabled />
            </DeprecatedMenu.Group>
          </DeprecatedMenu.List>
        </DeprecatedMenu.Popover>
      </DeprecatedMenu>
    )
  },
}

export const WithCompleteFeatures: StoryObj = {
  render: () => {
    return (
      <DeprecatedMenu>
        <DeprecatedMenu.Trigger>
          {({ getTriggerProps }) => (
            <DeprecatedButton {...getTriggerProps()} iconLeft={<DeprecatedIcon icon="more" fontSize="1rem" />} />
          )}
        </DeprecatedMenu.Trigger>
        <DeprecatedMenu.Popover>
          <DeprecatedMenu.List maxWidth="--size-80" maxHeight="--size-80">
            <DeprecatedMenu.Group label="Long Group Title that doesn't fit in one line">
              <DeprecatedMenu.Item
                label="Menu Item"
                supplementaryInfo="Short description about the item"
                badge={<Badge colour="neutral">Badge</Badge>}
                onClick={console.log}
                leftIcon={<DeprecatedIcon icon="property" />}
                rightIcon={<DeprecatedIcon icon="exportIcon" />}
              />
              <DeprecatedMenu.Item
                label="Menu Item anchor with long example text"
                supplementaryInfo="Secondary info long description that wonâ€™t fit in one line"
                badge={<Badge colour="neutral">Badge</Badge>}
                href="/#"
                leftIcon={<DeprecatedIcon icon="property" />}
                rightIcon={<DeprecatedIcon icon="exportIcon" />}
              />
              <DeprecatedMenu.Item
                label="Menu Item active"
                supplementaryInfo="Short description about the item"
                badge={<Badge colour="neutral">Badge</Badge>}
                isActive
                onClick={console.log}
                leftIcon={<DeprecatedIcon icon="property" />}
                rightIcon={<DeprecatedIcon icon="exportIcon" />}
              />
              <DeprecatedMenu.Item
                label="Menu Item (disabled)"
                supplementaryInfo="Short description about the item"
                badge={<Badge colour="neutral">Badge with long text</Badge>}
                onClick={console.log}
                disabled
                leftIcon={<DeprecatedIcon icon="property" />}
                rightIcon={<DeprecatedIcon icon="exportIcon" />}
              />
            </DeprecatedMenu.Group>
            <DeprecatedMenu.Group maxHeight="--size-32" label="Group Title">
              <DeprecatedMenu.Item label="Menu Item 1" onClick={console.log} />
              <DeprecatedMenu.Item label="Menu Item 2" href="/#" />
              <DeprecatedMenu.Item label="Menu Item 3" href="/#" />
              <DeprecatedMenu.Item label="Menu Item 4" href="/#" />
              <DeprecatedMenu.Item label="Menu Item 5" href="/#" />
              <DeprecatedMenu.Item label="Menu Item 6" href="/#" />
              <DeprecatedMenu.Item label="Menu Item 7" href="/#" />
            </DeprecatedMenu.Group>
          </DeprecatedMenu.List>
        </DeprecatedMenu.Popover>
      </DeprecatedMenu>
    )
  },
}

export const WithCustomAlignment: Story = {
  render: () => {
    return (
      <FlexContainer isFlexAlignCenter isFlexJustifyCenter>
        <DeprecatedMenu data-alignment="right">
          <DeprecatedMenu.Trigger>
            {({ getTriggerProps }) => (
              <DeprecatedButton {...getTriggerProps()} iconLeft={<DeprecatedIcon icon="more" fontSize="1rem" />} />
            )}
          </DeprecatedMenu.Trigger>
          <DeprecatedMenu.Popover yOffset={10}>
            <DeprecatedMenu.List>
              <DeprecatedMenu.Group label="Group Title">
                <DeprecatedMenu.Item label="Menu Item" onClick={console.log} />
                <DeprecatedMenu.Item label="Menu Item as anchor" href="/#" />
                <DeprecatedMenu.Item label="Menu Item (keep open)" closeMenu={false} />
              </DeprecatedMenu.Group>
            </DeprecatedMenu.List>
          </DeprecatedMenu.Popover>
        </DeprecatedMenu>
      </FlexContainer>
    )
  },
}

export const MoreComplexUsageExample: Story = {
  render: (props) => {
    const NavDropdownButtonUsageExample = ({
      title,
      yOffset,
      ...props
    }: ComponentProps<typeof DeprecatedMenu> & { title?: string; yOffset?: number }) => {
      return (
        <DeprecatedMenu {...props}>
          <DeprecatedMenu.Trigger>
            {({ getTriggerProps }) => (
              <DeprecatedButton
                {...getTriggerProps()}
                {...props}
                iconRight={<DeprecatedIcon icon="more" fontSize="1rem" />}
              >
                More
              </DeprecatedButton>
            )}
          </DeprecatedMenu.Trigger>
          <DeprecatedMenu.Popover yOffset={yOffset}>
            <DeprecatedMenu.List>
              <DeprecatedMenu.Group label={title ?? 'Group Title'}>
                <DeprecatedMenu.Item label="Menu Item" onClick={console.log} />
                <DeprecatedMenu.Item label="Menu Item as anchor" href="/#" />
                <DeprecatedMenu.Item label="Menu Item (keep open)" closeMenu={false} />
              </DeprecatedMenu.Group>
            </DeprecatedMenu.List>
          </DeprecatedMenu.Popover>
        </DeprecatedMenu>
      )
    }
    return (
      <FlexContainer
        isFlexColumn
        isFlexJustifyBetween
        style={{
          height: '150vh',
          overflow: 'hidden',
          padding: 5,
        }}
      >
        <FlexContainer isFlexJustifyBetween>
          <NavDropdownButtonUsageExample {...props} />
          <NavDropdownButtonUsageExample {...props} data-alignment="right" title="Custom y-offset" yOffset={50} />
        </FlexContainer>
        <FlexContainer isFlexJustifyBetween>
          <NavDropdownButtonUsageExample {...props} />
          <NavDropdownButtonUsageExample {...props} data-alignment="right" />
        </FlexContainer>
        <FlexContainer isFlexJustifyBetween>
          <NavDropdownButtonUsageExample {...props} />
          <NavDropdownButtonUsageExample {...props} data-alignment="right" />
        </FlexContainer>
      </FlexContainer>
    )
  },
}

export const WithIframe: Story = {
  render: () => {
    const IFrameComponent = () => {
      return (
        <iframe
          src="https://www.example.com"
          style={{
            width: '100%',
            height: '100%',
            border: 'none',
            marginTop: '1rem',
          }}
          title="Example IFrame"
        />
      )
    }

    // NOTE: make sure this component is rendered inside the Menu context
    const IframeHandler = () => {
      const { closeMenu } = useDeprecatedMenuContext()

      // close the menu when iframe is focused
      useEffect(() => {
        const controller = new AbortController()
        const handleWindowBlur = () => {
          if (document.activeElement?.tagName === 'IFRAME') {
            closeMenu()
          }
        }

        window.addEventListener('blur', handleWindowBlur, {
          signal: controller.signal,
        })
        return () => {
          controller.abort()
        }
      }, [closeMenu])
      return null
    }

    return (
      <MainContainer className={elHScreen}>
        <DeprecatedMenu>
          <DeprecatedMenu.Trigger>
            {({ getTriggerProps }) => (
              <DeprecatedButton {...getTriggerProps()} iconLeft={<DeprecatedIcon icon="more" fontSize="1rem" />} />
            )}
          </DeprecatedMenu.Trigger>
          <DeprecatedMenu.Popover>
            <IframeHandler />
            <DeprecatedMenu.List>
              <DeprecatedMenu.Group label="Group Title">
                <DeprecatedMenu.Item label="Menu Item" onClick={console.log} />
              </DeprecatedMenu.Group>
            </DeprecatedMenu.List>
          </DeprecatedMenu.Popover>
        </DeprecatedMenu>
        <IFrameComponent />
      </MainContainer>
    )
  },
}
