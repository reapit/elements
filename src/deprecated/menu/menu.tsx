import React, { type HTMLAttributes } from 'react'
import { DeprecatedMenuProvider } from './menu-context'
import { DeprecatedMenuPopover, DeprecatedMenuTrigger } from './menu-popover'
import { DeprecatedMenuItemGroup, DeprecatedMenuList } from './menu.atoms'
import { ElDeprecatedMenu } from './styles'
import { DeprecatedMenuItem } from './menu.molecules'

/** @deprecated */
type DeprecatedMenuFC = React.FC<HTMLAttributes<HTMLDivElement> & { 'data-alignment'?: 'left' | 'right' }> & {
  Item: typeof DeprecatedMenuItem
  Group: typeof DeprecatedMenuItemGroup
  Trigger: typeof DeprecatedMenuTrigger
  Popover: typeof DeprecatedMenuPopover
  List: typeof DeprecatedMenuList
}

/** @deprecated */
export const DeprecatedMenu: DeprecatedMenuFC = ({ children, ...props }) => {
  return (
    <DeprecatedMenuProvider>
      <ElDeprecatedMenu {...props}>{children}</ElDeprecatedMenu>
    </DeprecatedMenuProvider>
  )
}
DeprecatedMenu.Item = DeprecatedMenuItem
DeprecatedMenu.Group = DeprecatedMenuItemGroup
DeprecatedMenu.Trigger = DeprecatedMenuTrigger
DeprecatedMenu.Popover = DeprecatedMenuPopover
DeprecatedMenu.List = DeprecatedMenuList
