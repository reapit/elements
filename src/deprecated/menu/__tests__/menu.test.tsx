import { render } from '@testing-library/react'
import { DeprecatedMenu } from '../menu'

describe('Menu list component', () => {
  it('should render menu with list components and match snapshots', () => {
    const { asFragment } = render(
      <DeprecatedMenu>
        <DeprecatedMenu.List>
          <DeprecatedMenu.Group label="Group Title">
            <DeprecatedMenu.Item href="/#" label="Menu item as anchor" />
            <DeprecatedMenu.Item label="Menu item as button" />
          </DeprecatedMenu.Group>
        </DeprecatedMenu.List>
      </DeprecatedMenu>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
