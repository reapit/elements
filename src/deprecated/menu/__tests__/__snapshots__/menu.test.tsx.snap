// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu list component > should render menu with list components and match snapshots 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-1 el-deprecated-menu"
  >
    <div
      class="mocked-styled-12 el-deprecated-menu-list"
      data-has-max-width="false"
      role="menu"
      style="max-width: var(undefined); max-height: var(undefined);"
    >
      <div
        class="mocked-styled-10 el-deprecated-menu-item-group"
        role="group"
      >
        <div
          class="mocked-styled-9 el-deprecated-menu-item-group-title"
        >
          Group Title
        </div>
        <div
          class="mocked-styled-11 el-deprecated-menu-item-group-list"
          style="max-height: var(undefined);"
        >
          <a
            class="mocked-styled-8 el-deprecated-menu-item-anchor"
            data-close-menu="true"
            href="/#"
            role="menuitem"
          >
            <div
              class="mocked-styled-5 el-deprecated-menu-item-content"
            >
              <span
                class="mocked-styled-3 el-deprecated-menu-item-label-container"
              >
                <span
                  class="mocked-styled-4 el-deprecated-menu-item-label"
                >
                  Menu item as anchor
                </span>
              </span>
            </div>
          </a>
          <button
            class="mocked-styled-7 el-deprecated-menu-item-button"
            data-close-menu="true"
            role="menuitem"
            tabindex="0"
          >
            <div
              class="mocked-styled-5 el-deprecated-menu-item-content"
            >
              <span
                class="mocked-styled-3 el-deprecated-menu-item-label-container"
              >
                <span
                  class="mocked-styled-4 el-deprecated-menu-item-label"
                >
                  Menu item as button
                </span>
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
